core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/DefaultEncodingFactory.java

```diff
@@ -114,6 +114,9 @@ private ColumnPageEncoder createEncoderForDimensionLegacy(TableSpec.DimensionSpe
 
   private ColumnPageEncoder createEncoderForMeasureOrNoDictionaryPrimitive(ColumnPage columnPage,
       TableSpec.ColumnSpec columnSpec) {
+    if (columnPage.getDataType() == DataTypes.BINARY) {
```

Comment:
merge to “if (dataType == DataTypes.BOOLEAN) {”

Reply:
merged it to byte array